webpack/ForemanInventoryUpload/Components/PageHeader/__tests__/PageHeader.test.js

```diff
@@ -1,13 +1,114 @@
-import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
-
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { rtlHelpers } from 'foremanReact/common/rtlTestHelpers';
 import PageHeader from '../PageHeader';
+import { useAdvisorEngineConfig } from '../../../../common/Hooks/ConfigHooks';
+
+// Mock the config hook
+jest.mock('../../../../common/Hooks/ConfigHooks', () => ({
+  useAdvisorEngineConfig: jest.fn(),
+}));
+
+const { renderWithStore } = rtlHelpers;
 
-const fixtures = {
-  'render without Props': {},
-  /** fixtures, props for the component */
-};
+// Mock child components to isolate PageHeader testing
+// This prevents child component complexity from affecting our tests
+jest.mock('../components/SettingsWarning', () => () => (
+  <div data-testid="settings-warning">SettingsWarning</div>
+));
+jest.mock('../PageTitle', () => () => (
+  <div data-testid="page-title">PageTitle</div>
+));
+jest.mock('../../InventorySettings', () => () => (
+  <div data-testid="inventory-settings">InventorySettings</div>
+));
+jest.mock('../components/PageDescription', () => () => (
+  <div data-testid="page-description">PageDescription</div>
+));
+jest.mock('../../InventoryFilter', () => () => (
+  <div data-testid="inventory-filter">InventoryFilter</div>
+));
+jest.mock('../components/ToolbarButtons', () => () => (
+  <div data-testid="toolbar-buttons">ToolbarButtons</div>
+));
 
 describe('PageHeader', () => {
-  describe('rendering', () =>
-    testComponentSnapshotsWithFixtures(PageHeader, fixtures));
+  describe('component behavior', () => {
+    test('renders all components when not in IoP mode', () => {
+      useAdvisorEngineConfig.mockReturnValue(false);
+
+      renderWithStore(<PageHeader />);
```

Comment:
Instead of using the mock, you can render the component with the augmented state:

```suggestion
      renderWithStore(
        <PageHeader />,
        {
          "ADVISOR_ENGINE_CONFIG": {
            response: {
              use_iop_mode: false,
            },
          },
        });
```
This will implicitly test the `useAdvisorEngineConfig` accessor too, which is desired.

Reply:
@MariaAga as a general idea: should we add a helper method to construct API responses for tests?
Just to hide the internal structure of our `useAPI` implementation.
Something that will look like:

``` js
      renderWithStore(
        <PageHeader />,
        withAPIResponse(
          'ADVISOR_ENGINE_CONFIG',
          {
            use_iop_mode: false,
          }),
      );
```
The whole tree of the API response is not something that the plugin devs should be aware of. They only know the key if the request, and the response they expect to exist in the store.

Reply:
That could be a nice addition I agree